adc/ade7913/ade7913-driver.c

```diff
@@ -0,0 +1,753 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT1170 ADE7913 test application
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Marcin Baran
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+
+#include <sys/file.h>
+#include <sys/mman.h>
+#include <sys/msg.h>
+#include <sys/interrupt.h>
+#include <sys/platform.h>
+#include <sys/stat.h>
+#include <sys/threads.h>
+
+#include <imxrt-multi.h>
+#ifdef TARGET_IMXRT1170
+#include <phoenix/arch/imxrt1170.h>
+#endif
+
+#include <edma.h>
+
+#include "ade7913.h"
+#include "adc-api.h"
+
+#define COL_RED    "\033[1;31m"
+#define COL_CYAN   "\033[1;36m"
+#define COL_NORMAL "\033[0m"
+
+#define LOG_TAG "ade7913: "
+
+#ifdef NDEBUG
+#define log_debug(fmt, ...)
+#else
+#define log_debug(fmt, ...) \
+	do { \
+		printf(LOG_TAG fmt "\n", ##__VA_ARGS__); \
+	} while (0)
+#endif
+
+#define log_info(fmt, ...) \
+	do { \
+		printf(LOG_TAG COL_CYAN fmt COL_NORMAL "\n", ##__VA_ARGS__); \
+	} while (0)
+#define log_error(fmt, ...) \
+	do { \
+		printf(LOG_TAG COL_RED fmt COL_NORMAL "\n", ##__VA_ARGS__); \
+	} while (0)
+
+
+/*
+ * The buffer needs to be aligned with:
+ * num_of_devices * bytes_per_device * num_of_buffers
+ * so that in each filled buffer there will be same amount
+ * of samples for each device.
+ */
+#define ADC_BUFFER_SIZE             (6 * 16 * _PAGE_SIZE)
```

Comment:
That's quite a lot (49152 bytes). Does it have to be this big?

Reply:
You are right, I will change that.

Reply:
Changed to 16*_PAGE_SIZE (8192 bytes)